File: backend/F1Fantasy/Models/Driver.cs
namespace F1Fantasy.Models;

public class Driver
{
    public string DriverId { get; set; } = string.Empty;
    public string PermanentNumber { get; set; } = string.Empty;
    public string Code { get; set; } = string.Empty;
    public string Url { get; set; } = string.Empty;
    public string GivenName { get; set; } = string.Empty;
    public string FamilyName { get; set; } = string.Empty;
    public string DateOfBirth { get; set; } = string.Empty;
    public string Nationality { get; set; } = string.Empty;
    public List<string> ActiveSeasons { get; set; } = new List<string>();
}

// API Response models for Drivers endpoint
public class DriverApiResponse
{
    public DriverMRData? MRData { get; set; }
}

public class DriverMRData
{
    public string? Xmlns { get; set; }
    public string? Series { get; set; }
    public string? Url { get; set; }
    public string? Limit { get; set; }
    public string? Offset { get; set; }
    public string? Total { get; set; }
    public DriverTable? DriverTable { get; set; }
}

public class DriverTable
{
    public string? Season { get; set; }
    public List<Driver>? Drivers { get; set; }
}

File: backendJava/f1fantasy/src/main/java/no/f1fantasy/entity/Driver.java
package no.f1fantasy.entity;

import jakarta.persistence.*;
import lombok.Data;
import no.f1fantasy.entity.converter.StringArrayConverter;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "Drivers")
@Data
public class Driver {

    @Id
    @Column(name = "DriverId", length = 100)
    private String driverId;

    @Column(name = "PermanentNumber", length = 10)
    private String permanentNumber;

    @Column(name = "Code", length = 10)
    private String code;

    @Column(name = "Url", length = 500)
    private String url;

    @Column(name = "GivenName", length = 100)
    private String givenName;

    @Column(name = "FamilyName", length = 100)
    private String familyName;

    @Column(name = "DateOfBirth", length = 50)
    private String dateOfBirth;

    @Column(name = "Nationality", length = 100)
    private String nationality;

    /** Stored as {@code text[]} in PostgreSQL, as a JSON-like array string in H2. */
    @Convert(converter = StringArrayConverter.class)
    @Column(name = "ActiveSeasons")
    private List<String> activeSeasons = new ArrayList<>();
}
